### PR TITLE
Support Python 3.13

### PR DESCRIPTION
**Pull request by jameshilliard**
# Summary

Run tests on 3.13 and update version support.

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.

### CODE REVIEW

**Comment by zanieb** [comment]
Looks like there's a problem with `uvloop`

**Comment by jameshilliard** [comment]
> Looks like there's a problem with `uvloop`

Yeah, had to set a minimum uvloop version for python 3.13, `httptools` also had an issue with using the wrong cython version.

**Comment by Kludex** [comment]
I'd prefer for the httptools and uvloop to make a release before proceeding here.

**Comment by jameshilliard** [comment]
httptools now has a release with 3.13 support

**Comment by Kludex** [comment]
Why is uvloop in beta?

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +42,8 @@ standard = [
     "httptools>=0.5.0",
     "python-dotenv>=0.13",
     "PyYAML>=5.1",
-    "uvloop>=0.14.0,!=0.15.0,!=0.15.1; sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')",
+    "uvloop>=0.14.0,!=0.15.0,!=0.15.1; python_version < '3.13' and sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')",
+    "uvloop>=0.21.0b1; python_version >= '3.13' and sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')",
```
Comment (Kludex) [inline pyproject.toml]:
```suggestion
    "uvloop>=0.14.0,!=0.15.0,!=0.15.1; sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')",
```

**Review by Kludex — APPROVED** [review]

**Pull request merged**